addons/project/models/project.py

```diff
@@ -915,6 +920,8 @@ def action_assign_to_me(self):
         self.write({'user_id': self.env.user.id})
 
     def action_open_parent_task(self):
+        if self.sudo().parent_id and self.sudo().parent_id.company_id.id not in self.env.companies.ids:
+            raise UserError(_('The parent task belongs to a company you do not have access to.'))
```

Comment:
@tivisse @jdoutreloux @aju-odoo 
I just had to remove this diff in our production when we upgraded to 13.0,
as well as the one in `def action subtask`.
Basically, Odoo DMCC (Dubai) orders subtasks to our PS services in Belgium,
(they request custom developments in the project PS saas customization)
and they therefore creates subtasks in Odoo BE on tasks in Odoo DMCC.
It's a use case which makes sense in our company.

I don't think we want this constraint in our database, and even if we do, it's weird to put such a constraint in a smart button. Usually, such a constraint is done thanks to a record rule, or an sql constraint. I can see that by default in Odoo 13.0 there is the record rule *Project/Task: multi-company*:
```python
[('company_id', 'in', company_ids)]
```
Isn't that enough? Thanks to this you shoudln't be able to assign as parent task a task from a different company than yours as you will not see it. Besides, if you want to disable this in your database (such as in our database), you just have to disable the record rule, and you don't need a local diff to remove this constraint.

Reply:
Hi @beledouxdenis 

Thanks for the feedback.

This has been introduced because the product owner wanted to display an user friendly error message instead of the generic one coming from the ir.rule.

But that's true, if you relax the ir.rule, the error would still occur from the code you spotted. 

It makes sense to remove this chunk of code then.

Reply:
Is someone taking care of this or should I do it myself?

Reply:
There is a task in our project to handle this. See: https://www.odoo.com/web#id=2198433&action=333&active_id=1251&model=project.task&view_type=form&cids=1&menu_id=4720

But if you prefer to do it yourself it's ok ;)

Reply:
As long it's in the pipe it's ok, I just wanted to make sure it was being taken care.